website/src/app/api.service.ts

```diff
@@ -0,0 +1,73 @@
+import { Injectable } from '@angular/core';
+import { HttpModule, Http } from '@angular/http';
+import 'rxjs/Rx';
+import { Observable } from 'rxjs/Observable';
+import { User } from './user'
+import { environment } from 'src/environments/environment';
+
+const API_URL = environment.apiUrl;
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ApiService {
+  public addUser(user: User): Observable<User> {
+    return this.http
+    .post(API_URL + '/users', user)
+    .map(response => {
+      return new User(response.json());
```

Comment:
This is not needed. You could just assume it returns `User` directly and indicate it by passing the type parameter like this `this.http.get<User>(...)`